### PR TITLE
MM-47171 Export direct channel favorites

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
Favorited direct channels were missing from the export file. This PR adds them. The import part was already implemented.

#### Ticket Link
[MM-47171](https://mattermost.atlassian.net/browse/MM-47171)

#### Release Note
```release-note
Exports now contain favorited direct channels
```

### CODE REVIEW

File: app/export.go [hunk 1]
```diff
@@ -636,6 +641,29 @@ func (a *App) exportAllDirectChannels(writer io.Writer) *model.AppError {
 	return nil
 }
 
+func (a *App) buildFavoritedByList(channelID string) ([]string, *model.AppError) {
+	prefs, err := a.Srv().Store().Preference().GetCategoryAndName(model.PreferenceCategoryFavoriteChannel, channelID)
+	if err != nil {
+		return nil, model.NewAppError("buildFavoritedByList", "app.channel.get_all_direct.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```
Comment (isacikgoz) [inline app/export.go:647]: The error string is a little out of context I guess. 
```suggestion
		return nil, model.NewAppError("buildFavoritedByList", "app.preference.get_category.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```

File: app/export.go [hunk 2]
```diff
@@ -636,6 +641,29 @@ func (a *App) exportAllDirectChannels(writer io.Writer) *model.AppError {
...
+
+	userIDs := make([]string, 0, len(prefs))
+	for _, pref := range prefs {
+		if pref.Value != "true" {
+			continue
+		}
+
+		user, err := a.Srv().Store().User().Get(context.Background(), pref.UserId)
+		if err != nil {
+			return nil, model.NewAppError("buildFavoritedByList", "app.channel.get_all_direct.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```
Comment (isacikgoz) [inline app/export.go:658]:
```suggestion
			return nil, model.NewAppError("buildFavoritedByList", "app.user.get.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```

**Review by isacikgoz — APPROVED** [review]
LGTM, two small suggestions.

**Comment by noxer** [comment]
Good points @isacikgoz, thanks!

File: store/sqlstore/preference_store.go [hunk 3]
```diff
@@ -140,6 +140,23 @@ func (s SqlPreferenceStore) Get(userId string, category string, name string) (*m
 	return &preference, nil
 }
 
+func (s SqlPreferenceStore) GetCategoryAndName(category string, name string) (model.Preferences, error) {
```
Comment (agnivade) [inline store/sqlstore/preference_store.go:143]: This should have been `GetByCategoryAndName` as that's how we do it elsewhere, but I do see that we have a `GetCategory` already. So good for now.

**Review by agnivade — APPROVED** [review]

**Pull request merged**